lib/partitionMessages.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2014 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+"use strict";
+
+var utils = require('./utils');
+
+/**
+ * PartitionMessages resource, scoped to a single Partition.
+ * @type {Function}
+ */
+var PartitionMessages = module.exports = function(partition) {
+    this.client = partition.client;
+    this.partition = partition;
+}
+
+/**
+ * Request a specified number of messages from a specified offset
+ *
+ * type options {
+ *      offset: number,
+ *      count: ?number,
+ *      format: ?string (json, avro, binary)
+ *  }
+ *
+ * @param options
+ * @param res function(err, messages)
+ */
+PartitionMessages.prototype.messages = function(options, res) {
+    options = options || {};
+
+    // Ensure that offset is passed in
+    if (typeof options === 'function' || options.offset === undefined || options.offset === null) {
+        var callback = res || options;
+        return callback(new Error('Offset is required'));
+    }
+
+    var query = { offset: options.offset };
+
+    // Count is optional
+    if (options.count) query.count = options.count;
+
+    var reqOpts = {
+        path: this.getPath() + this.getQueryString(query),
+        accept: this.getContentType(options.format)
+    };
+
+    this.client.request(reqOpts, function(err, messages_raw) {
+        if (err) return res(err);
+        this.raw = messages_raw;
+        res(null, this);
+    }.bind(this));
+}
+
+PartitionMessages.prototype.getPath = function() {
+    return utils.urlJoin(this.partition.getPath(), "messages");
+}
+
+PartitionMessages.prototype.getQueryString = function(query) {
+    return utils.urlQueryString(query);
+}
+
+PartitionMessages.prototype.getContentType = function(format) {
+    var version = this.client.config.version;
+
+    switch (format) {
```

Comment:
I think I see why this was a problem now. In the regular high-level consumer code, I use a schema class, but there's just some manual parsing to get from a format string like `avro` or `binary` to the appropriate `Schema` class. This is not very nice right now, but worked ok for awhile because we only supported 2 formats -- it's a bit less obvious when it only requires one `if` statement rather than a `switch`. You're also supporting JSON here, which is complicating things further.

I think this is fine for now -- we'll just want to revamp how schemas are used at some point to make this all cleaner.